gateway/modules/schema/helpers.go

```diff
@@ -401,62 +440,49 @@ type indexStruct struct {
 	IndexMap      []*model.FieldType
 }
 
-func getRealIndexMap(realTableInfo model.Fields) (map[string]*indexStruct, error) {
-	realIndexMap := make(map[string]*indexStruct)
-	for _, realColumnInfo := range realTableInfo {
-		if realColumnInfo.IsIndex {
-			if value, ok := realIndexMap[realColumnInfo.IndexInfo.Group]; ok {
-				value.IndexMap = append(value.IndexMap, realColumnInfo)
-			} else {
-				realIndexMap[realColumnInfo.IndexInfo.Group] = &indexStruct{IndexMap: []*model.FieldType{realColumnInfo}}
-			}
-			if realColumnInfo.IsUnique {
-				realIndexMap[realColumnInfo.IndexInfo.Group].IsIndexUnique = true
-			}
-			if !(realColumnInfo.IndexInfo.Sort == "asc" || realColumnInfo.IndexInfo.Sort == "desc") {
-				return nil, errors.New("Invalid Sort")
+func getIndexMap(tableInfo model.Fields) (map[string]*indexStruct, error) {
+	indexMap := make(map[string]*indexStruct)
+
+	// Iterate over each column of table
+	for _, columnInfo := range tableInfo {
+
+		// We are only interested in the columns which have an index on them
+		if columnInfo.IsIndex {
+
+			// Append the column to te index map. Make sure we create an empty array if no index by the provided name exists
+			value, ok := indexMap[columnInfo.IndexInfo.Group]
+			if !ok {
+				value = &indexStruct{IndexMap: []*model.FieldType{}}
+				indexMap[columnInfo.IndexInfo.Group] = value
 			}
-		}
-	}
+			value.IndexMap = append(value.IndexMap, columnInfo)
 
-	for _, indexValue := range realIndexMap {
-		var v indexStore = indexValue.IndexMap
-		sort.Stable(v)
-		indexValue.IndexMap = v
-		for i, column := range indexValue.IndexMap {
-			if i+1 != column.IndexInfo.Order {
-				return nil, errors.New("Index Order Invalid")
+			// Mark the index group as unique if even on column had the unique tag
+			if columnInfo.IsUnique {
+				indexMap[columnInfo.IndexInfo.Group].IsIndexUnique = true
 			}
-		}
-	}
-	return realIndexMap, nil
-}
 
-func getCurrentIndexMap(currentTableInfo model.Fields) (map[string]*indexStruct, error) {
-	currentIndexMap := make(map[string]*indexStruct)
-	for _, currentColumnInfo := range currentTableInfo {
-		if currentColumnInfo.IsIndex {
-			if value, ok := currentIndexMap[currentColumnInfo.IndexInfo.Group]; ok {
-				value.IndexMap = append(value.IndexMap, currentColumnInfo)
-			} else {
-				currentIndexMap[currentColumnInfo.IndexInfo.Group] = &indexStruct{IndexMap: []*model.FieldType{currentColumnInfo}}
+			// Set the default sorting scheme if not present
+			if columnInfo.IndexInfo.Sort == "" {
+				columnInfo.IndexInfo.Sort = "asc"
 			}
-			if currentColumnInfo.IsUnique {
-				currentIndexMap[currentColumnInfo.IndexInfo.Group].IsIndexUnique = true
+
+			// Make sure the sort type is either ascending or descending
+			if !(columnInfo.IndexInfo.Sort == "asc" || columnInfo.IndexInfo.Sort == "desc") {
+				return nil, fmt.Errorf("invalid sort type (%s) provided for field (%s)", columnInfo.IndexInfo.Sort, columnInfo.FieldName)
```

Comment:
can we do this in the parse function of schema module ?

Reply:
I don't know. This was already here. I'll move it somewhere within SetConfig